src/processors/modulation/Flanger.cpp

```diff
@@ -0,0 +1,298 @@
+    }
+
+    // set phase offsets
+    constexpr float piOver3 = MathConstants<float>::pi / 3.0f;
+    LFOs[0][0].reset (-piOver3);
+    LFOs[1][1].reset (piOver3);
+
+    aaFilter.prepare (spec);
+    aaFilter.setCutoffFrequency (12000.0f);
+
+    dryWetMixer.prepare (spec);
+    dryWetMixer.setMixingRule (dsp::DryWetMixingRule::sin3dB);
+
+    dcBlocker.prepare (spec);
+    dcBlocker.setCutoffFrequency (60.0f);
+
+    audioOutBuffer.setSize (2, samplesPerBlock);
+    modOutBuffer.setSize (1, samplesPerBlock);
+
+    for (auto& filt : hilbertFilter)
+        filt.reset();
+
+    bypassNeedsReset = false;
+}
+
+void Flanger::processModulation (int numSamples)
+{
+    modOutBuffer.setSize (1, numSamples, false, false, true);
+    if (inputsConnected.contains (ModulationInput))
+    {
+        // get modulation buffer from input (-1, 1)
+        const auto& modInputBuffer = getInputBuffer (ModulationInput);
+        BufferHelpers::collapseToMonoBuffer (modInputBuffer, modOutBuffer);
+
+        auto phaseShiftLFO = [this, numSamples, filterIndex = 0] (float* lfoIn, float* lfoOut) mutable
+        {
+            // trying to use more of the Hilbert filters than we have available!
+            jassert (filterIndex < (int) std::size (hilbertFilter));
+
+            auto& filter = hilbertFilter[filterIndex++];
+            for (int n = 0; n < numSamples; ++n)
+                std::tie (lfoIn[n], lfoOut[n]) = filter.process (lfoIn[n]);
+        };
+
+        auto* modOutData = modOutBuffer.getReadPointer (0);
+
+        FloatVectorOperations::copy (LFOData[0][0].data(), modOutData, numSamples);
+        phaseShiftLFO (LFOData[0][0].data(), LFOData[0][1].data());
+        FloatVectorOperations::copy (LFOData[1][0].data(), LFOData[0][1].data(), numSamples);
+        phaseShiftLFO (LFOData[1][0].data(), LFOData[1][1].data());
+    }
+    else
+    {
+        auto rate = rateLow * std::pow (rateHigh / rateLow, *rateParam);
+        for (int ch = 0; ch < 2; ++ch)
+        {
+            for (int i = 0; i < delaysPerChannel; ++i)
+            {
+                LFOs[ch][i].setFrequency (rate);
+            }
+
+            for (int i = 0; i < delaysPerChannel; ++i)
+            {
+                auto* data = LFOData[ch][i].data();
+                for (int n = 0; n < numSamples; ++n)
+                    data[n] = LFOs[ch][i].processSample();
+            }
+        }
+
+        auto* modOutData = modOutBuffer.getWritePointer (0);
+        FloatVectorOperations::copy (modOutData, LFOData[0][0].data(), numSamples);
+    }
+}
+
+template <typename DelayArrType>
+void Flanger::processFlanger (AudioBuffer<float>& buffer, DelayArrType& delay)
+{
+    jassert (buffer.getNumChannels() == 2); // always processing in stereo
+    const auto numSamples = buffer.getNumSamples();
+
+    for (int ch = 0; ch < 2; ++ch)
+    {
+        for (int i = 0; i < delaysPerChannel; ++i)
+            delay[ch][i].setFilterFreq (10000.0f);
```

Comment:
I've seen some flangers with a resonance control, would that add anything cool to this one?

Reply:
Maybe... I'm not sure it would be super useful here since the filter frequency isn't available as a user parameter.